src/sites/experiments.cubing.net/cubing.js/acube/index.ts

```diff
@@ -0,0 +1,203 @@
+import { puzzles, PuzzleLoader } from "../../../../cubing/puzzles";
+
+interface Piece {
+  [orientation: number]: Facelet;
+}
+type Mode = 'swap' | 'twist';
+
+class App {
+  mode: Mode = 'swap';
+  cube: Cube;
+
+  constructor() {
+    const puzzleSelect = document.querySelector(
+      "#puzzles",
+    ) as HTMLSelectElement;
+
+    for (const puzzle in puzzles) {
+      const option: HTMLOptionElement = puzzleSelect.appendChild(
+        document.createElement("option"),
+      )!;
+      option.value = puzzle;
+      option.textContent = puzzles[puzzle].fullName;
+    }
+
+    const puzzle = new URL(location.href).searchParams.get("puzzle") || '3x3x3';
+    if (puzzle) {
+      if (puzzle in puzzles) {
+        this.cube = new Cube(puzzles[puzzle]);
+        puzzleSelect.value = puzzle;
+      } else {
+        console.error("Invalid puzzle:", puzzle);
+      }
+    }
+
+    puzzleSelect?.addEventListener("change", () => {
+      this.cube.setPuzzle(puzzles[puzzleSelect.value]);
+
+      const url = new URL(location.href);
+      url.searchParams.set("puzzle", puzzleSelect.value);
+      window.history.replaceState("", "", url.toString());
+    });
+
+    document.querySelectorAll('input[name="mode"]').forEach((radio: HTMLInputElement) => {
+      radio.addEventListener('change', () => {
+        this.mode = radio.value as Mode;
+      });
+    });
+  }
+}
+
+class Cube {
+  selectedFacelet: Facelet | null;
+  pieces = new Map();
+
+  puzzle: PuzzleLoader;
+  
+  constructor(puzzle: PuzzleLoader) {
+    this.setPuzzle(puzzle);
+  }
+
+  setPuzzle(puzzle: PuzzleLoader) {
+    this.puzzle = puzzle;
+    
+    puzzle.svg().then(svg => {
+      document.querySelector("#puzzle")!.innerHTML = svg;
+      document.querySelector('svg')!.removeAttribute('width');
```

Comment:
FTO had width and height hardcoded on the SVG.  Ideally these lines aren't needed however for now it fixes the FTO issue.

Reply:
Ideally we'd handle that in CSS, but that that sounds like an okay hack just to get things working initially.